src/webgpu/web-platform/reftests/canvas_complex.ts

```diff
@@ -0,0 +1,78 @@
+import { unreachable, assert } from '../../../common/framework/util/util.js';
+
+import { runRefTest } from './gpu_ref_test.js';
+
+// <canvas> element from html page
+declare const cvs: HTMLCanvasElement;
+
+const formats = new URLSearchParams(window.location.search).getAll('format');
+assert(formats.length === 1);
+const format = formats[0];
+
+runRefTest(async t => {
+  const ctx = (cvs.getContext('gpupresent') as unknown) as GPUCanvasContext;
+
+  switch (format) {
+    case 'bgra8unorm':
+    case 'rgba16float':
+      break;
+    default:
+      unreachable();
+  }
+
+  const swapChain = ctx.configureSwapChain({
+    device: t.device,
+    format,
+    usage: GPUTextureUsage.COPY_DST,
+  });
+
+  const rows = 2;
+  const bytesPerRow = 256;
+  const [buffer, mapping] = t.device.createBufferMapped({
+    size: rows * bytesPerRow,
+    usage: GPUBufferUsage.COPY_SRC,
+  });
+  switch (format) {
```

Comment:
A quick(silly) question, No rgba8unorm?

Reply:
That's right. Initially I did test it, but actually I decided to disallow it in Chrome, as it's not supposed to be supported by default. (bgra8unorm and rgba16float were the two formats we tentatively agreed on for core.)